Sources/X509/OCSP/OCSPPolicy.swift

```diff
@@ -31,8 +31,50 @@ public protocol OCSPRequester: Sendable {
     /// - Parameters:
     ///   - request: DER-encoded request bytes
     ///   - uri: uri of the OCSP responder
-    /// - Returns: DER-encoded response bytes
-    func query(request: [UInt8], uri: String) async throws -> [UInt8]
+    /// - Returns: DER-encoded response bytes if they request was successful or a terminal or non-terminal error.
+    func query(request: [UInt8], uri: String) async -> OCSPRequesterQueryResult
+}
+
+public struct OCSPRequesterQueryResult: Sendable {
+    @usableFromInline
+    enum Storage: Sendable {
+        case success([UInt8])
+        case nonTerminal(Error)
+        case terminal(Error)
+    }
+    @usableFromInline
+    var storage: Storage
+    
+    @inlinable
+    init(_ storage: Storage) {
+        self.storage = storage
+    }
+}
+
+extension OCSPRequesterQueryResult {
+    /// The OCSP query is considered successful and has returned the given DER-encoded response bytes.
+    /// - Parameter bytes: DER-encoded response bytes
+    @inlinable
+    public static func response(_ bytes: [UInt8]) -> Self {
+        .init(.success(bytes))
+    }
+    
+    /// The OCSP query is considered unsuccessful but will **not** fail verification, neither in ``OCSPFailureMode/soft`` nor in ``OCSPFailureMode/hard`` failure mode.
+    /// The certificate is then considered to meet the ``OCSPVerifierPolicy``.
+    /// - Parameter reason: the reason why the OCSP query failed which may be used for diagnostics
+    @inlinable
+    public static func nonTerminalError(_ reason: Error) -> Self {
```

Comment:
The doc string here should probably have a `warning` directive that calls out that if you use this the OCSP validation will be assumed to have _succeeded_.

Reply:
Added in https://github.com/apple/swift-certificates/pull/108/commits/8c659ed5947b6de8886ee8bbb31b514aac13ea29